### PR TITLE
Adds dev observability section

### PR DESCRIPTION
**Pull request by v-bpastiu**
- adds dev observability section
- metrics server, kubernetes dashboard to be used for development environment observability

### CODE REVIEW

File: docs/02-development/development-observability.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+See details about the resources that have been allocated to your nodes, rather than the current resource usage, the kubectl describe command provides a detailed breakdown of a specified pod or node.
+
+```shell
+kubectl describe node <NODE_NAME>
+```
+
+The output is quite verbose containing a full breakdown of the node’s workloads, system info, and metadata such as labels and annotations.
+
+??? note "Click to expand `node desribe command` output"
```
Comment (v-ctiutiu) [inline docs/02-development/development-observability.md:89]: Typo: `desribe` -> `describe`

**Review by v-ctiutiu — CHANGES_REQUESTED** [review]

File: docs/02-development/development-observability.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        --------           --------          ------
+        cpu                1382m (35%)       1227m (31%)
+        memory             1721869056 (24%)  1970381568 (28%)
+        ephemeral-storage  0 (0%)            0 (0%)
+        hugepages-1Gi      0 (0%)            0 (0%)
+        hugepages-2Mi      0 (0%)            0 (0%)
+
+## Installing the Kubernetes Dashboard
+
+In this section you will install the community maintained [Kubernetes Dashboard](https://github.com/kubernetes/dashboard). Please follow below steps to install it using kubectl:
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:189]: Fixed in latest commit.

File: docs/02-development/development-observability.md (hunk 1 above)
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:89]: Fixed in latest commit.

**Review by v-ctiutiu — CHANGES_REQUESTED** [review]

File: mkdocs.yml [hunk 3]
```diff
@@ -17,8 +17,7 @@ nav:
       - Tilt for local development: 02-development/tilt-local.md
       - Tilt for remote development: 02-development/tilt-remote.md
       - Set up ingress: 02-development/setup-ingress.md
-      - Set up logging: 02-development/setup-logging.md
-      - Set up monitoring: 02-development/setup-monitoring.md
+      - Development observability: 02-development/development-observability.md
```
Comment (v-ctiutiu) [inline mkdocs.yml:20]: I would leave only `Observability` as the main title for this section.

File: docs/02-development/setup-ingress.md [hunk 4]
```diff
@@ -211,3 +211,8 @@ In this section you will install the community maintained version of the Nginx i
 11. Open a web browser and point to `<YOUR_A_RECORD>.<YOUR_DOMAIN>`. You should see the online boutique welcome page. The connection is secure and the certificate is a valid one issued by [Let's Encrypt](https://letsencrypt.org).
 
     ![development environment online boutique](microservices_demo_ingress_dev.png)
+
+!!! note
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:215]: This note should be moved to the Observability section no ?

And I would rephrase like this:

>You will not install a full blown monitoring and logging stack on the development cluster such as Prometheus and Loki, but will do so for the `staging` and `production` clusters.

File: docs/02-development/development-observability.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+    ```shell
+    helm repo add metrics-server https://kubernetes-sigs.github.io/metrics-server/
+    ```
+
+2. Install the `Kubernetes Metrics Server` using `Helm`:
+
+    ```shell
+    helm upgrade --install metrics-server metrics-server/metrics-server
+    ```
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:28]: done

File: docs/02-development/development-observability.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+emailservice-f8795cc94-r7hwp             30m          40Mi            
+frontend-6d45d8cc5d-59fmw                1m           20Mi            
+paymentservice-995d69494-kcqn2           5m           30Mi            
+productcatalogservice-556d4f9446-7sqp9   7m           16Mi            
+recommendationservice-59f78c445b-5487v   40m          40Mi            
+redis-cart-596c7658c4-lwf8g              3m           7Mi             
+shippingservice-bfc488696-dkcpz          3m           15Mi 
+```
+
+See details about the resources that have been allocated to your nodes, rather than the current resource usage, the kubectl describe command provides a detailed breakdown of a specified pod or node.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:81]: Done

File: docs/02-development/development-observability.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+    ```
+
+2. In a new terminal window start the `kubectl proxy`:
+
+    ```shell
+    kubectl proxy
+    ```
+
+3. Open a web browser and point to the [Kubernetes Dashboard Login](http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:https/proxy/#/login) and add your cluster's config file in the `Login` page.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:206]: Done

File: docs/02-development/development-observability.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+
+    !!! note
+            You will authenticate using the cluster's `kubeconfig` file. To get the config file navigate to your DigitalOcean cloud console, go to `Kubernetes`, select your cluster and from the `Configuration` section, click on `Download Config File`.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:209]: Done

File: docs/02-development/development-observability.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+    You should see the following:
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:211]: Done

File: docs/02-development/development-observability.md [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+
+    ![kubernetes dashboard landing page](kubernetes_dashboard_landing_page.png)
+
+Once you deploy and log in to Kubernetes Dashboard, you’ll have access to metric summaries for each pod, node, and namespace in your cluster. You can also use the UI to edit Kubernetes objects—for instance, to scale up a Deployment or to change the image version in a pod’s specification.'
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:215]: Done

File: docs/02-development/development-observability.md [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+
+You can also view the stream of logs from a pod in `Kubernetes Dashboard`. From the navigation bar at the top of the `Pods` view, click on the `Logs` tab to access a log stream from the pod in the browser, which can be further segmented by container if the pod comprises multiple containers. `Exec` into a container is also possible from the same page.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:217]: Done

File: docs/02-development/development-observability.md [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+
+Kuberentes events are also viewable from the `Kubernetes Dashboard`. From the left menu click on the `Events` view. Events will be displayed and stored for 1 hour.
+
+Next, you will setup the `staging` environment.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/development-observability.md:221]: Done

File: docs/02-development/setup-ingress.md [hunk 13]
```diff
@@ -211,3 +211,8 @@ In this section you will install the community maintained version of the Nginx i
...
+    You will not install monitoring, logging and alerting on the development cluster but will do so for the `staging` and `production` clusters.
+
+Next, you will deploy the [Kubernetes Dashboard](https://github.com/kubernetes/dashboard) and [Kubernetes Metrics Server](https://github.com/kubernetes-sigs/metrics-server) which will allow you to visualize application and cluster related metrics as well as logs and events.
```
[the remark this answers is not in the document]
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:218]: Done

File: mkdocs.yml (hunk 3 above)
Comment (v-bpastiu) [reply mkdocs.yml:20]: Done

File: docs/02-development/setup-ingress.md (hunk 4 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md]: Done

**Review by v-ctiutiu — APPROVED** [review]
Looks good to me.
